crates/bevy_app/src/app.rs

```diff
@@ -1171,6 +1183,13 @@ mod tests {
         assert_eq!(app.world.entities().len(), 2);
     }
 
+    #[test]
+    fn cannot_use_plugin_registry_during_finish() {
```

Comment:
Shouldn't this test be `should_panic` and call the `Plugin::finish` above?

Reply:
Oh I wasn't calling `app.finish()`, oops!

No, my point is that the test does not panic because `app.is_plugin_added` **always** returns false in `app.finish()` and `app.cleanup()`, which is kind of unintuitive..